### PR TITLE
ENH: Add softplus to paddle frontend

### LINKED ISSUE #15949: softplus

### LINKED ISSUE #18342: Softplus

### PR DESCRIPTION
**Pull request by aadya940**
Closes  #18342

### CODE REVIEW

**Comment by aadya940** [comment]
Can anyone let me know why the tests are failing?

**Comment by aadya940** [comment]
@Sam-Armstrong

**Comment by Sam-Armstrong** [comment]
Sorry, I've been really busy - will review over the weekend :)

File: ivy/functional/frontends/paddle/nn/functional/activation.py [hunk 1]
```diff
@@ -209,3 +209,8 @@ def elu(
 @to_ivy_arrays_and_back
 def mish(x, name=None):
     return ivy.mish(x)
+
```
Comment (Sam-Armstrong) [inline ivy/functional/frontends/paddle/nn/functional/activation.py:212]: we need two lines of whitespace between functions, not one

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_paddle_activation.py [hunk 2]
```diff
@@ -514,3 +514,30 @@ def test_paddle_mish(
...
     )
+
+
+@handle_frontend_test(
+    fn_tree="paddle.nn.functional.softplus",
+    dtype_and_input=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("valid"),
+        safety_factor_scale="log",
+        small_abs_safety_factor=20,
+    ),
```
Comment (Sam-Armstrong) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_paddle_activation.py:525]:
```suggestion
    dtype_and_input=helpers.dtype_and_values(
        available_dtypes=helpers.get_dtypes("valid"),
        min_num_dims=1,
    ),
```

**Review by Sam-Armstrong — CHANGES_REQUESTED** [review]
@aadya940 thanks for making the PR! I've made a couple of comments with small things to change (the second comment fixes the tests), but otherwise this looks really good. We should be good to merge once you've made these changes 😁

**Comment by aadya940** [comment]
Thank you @Sam-Armstrong :-)

**Review by Sam-Armstrong — APPROVED** [review]
lgtm! thanks @aadya940

**Pull request merged**